projects/vdk-plugins/vdk-impala/tests/functional/template_regression_test.py

```diff
@@ -417,9 +417,12 @@ def test_load_fact_snapshot_empty_source(self) -> None:
             },
         )
         # Expecting data job not to finish due to empty source.
-        assert res.exception
-        assert "Source view returned no results." in res.exception.args[0]
-        assert isinstance(res.exception, errors.UserCodeError)
+        assert not res.exception
+        assert "Source view returned no results." in res.output
+        assert (
```

Comment:
Let's not get used to verifying by logs. That's not a good practice as we can change the logs to make them more user friendly or changing their log level and their format is not fixed . It's a last option in case there's no other way. 

Another reason is we can log "execution was skipped" but it may not be skipped (maybe we have a bug and the steps are not actually skipped) . The log is misleading but the test is false positive. 

I think in this case checking that exist status is success and that there's no exception is  enough. The template with empty view has succeeded.